gravitee-apim-console-webui/src/management/api/general/user-group-access/members/api-general-group-members/api-general-group-members.component.ts

```diff
@@ -82,25 +82,32 @@ export class ApiGeneralGroupMembersComponent implements OnInit, OnDestroy {
 
   private getGroupMembersPage(page = 1, perPage = 10): void {
     this.groupService
-      .getMembers(this.groupId, page, perPage)
+      .getMembers(this.groupData.id, page, perPage)
       .pipe(takeUntil(this.unsubscribe$))
-      .subscribe((membersResponse) => {
-        if (!membersResponse.pagination.totalCount || membersResponse.pagination.totalCount === 0) {
-          this.destroy.emit();
-          return;
-        }
-        this.dataSourceGroup = {
-          groupName: membersResponse.metadata?.groupName,
-          memberTotalCount: membersResponse.pagination.totalCount,
-          dataSource: membersResponse.data.map((member) => {
-            return {
-              id: member.id,
-              role: member.roles.find((role) => role.scope === 'API')?.name,
-              displayName: member.displayName,
-              picture: this.userService.getUserAvatar(member.id),
-            };
-          }),
-        };
-      });
+      .subscribe(
```

Comment:
Would be nice to refactor this into smaller bitesize functions to reduce cognitive complexity
<img width="836" alt="Cognitive complexity of getGroupMembersPage function" src="https://github.com/gravitee-io/gravitee-api-management/assets/137767084/803ade12-8527-4bf7-8ecf-641e2d443430">


There are handy tools that can help point this out like https://marketplace.visualstudio.com/items?itemName=kisstkondoros.vscode-codemetrics if your a VS Code user

or via sonar cloud if that is available.

general rule I follow: When I start to write a  for, if/else, switch, while or a nested  loop then it's likely I'll refactor it out of the bigger function and call it from it's own function.

Although trivial:
`!membersResponse.pagination.totalCount || membersResponse.pagination.totalCount === 0`

this could be extracted into its own function with a more meaningful reference:
```
hasMembersInTheResponse(membersResponse): boolean {
return membersResponse.pagination.totalCount && membersResponse.pagination.totalCount !== 0
}
```
```
if (!this.hasMembersInTheResponse(membersResponse)) {
            this.destroy.emit();
            return;
          }
          ```
          Then you kind of read the code like a book.
          
some interesting reads:
https://blog.codacy.com/an-in-depth-explanation-of-code-complexity/
https://linearb.io/blog/what-is-code-complexity

However this is only my opinion. It may be too verbose in some cases.

Reply:
I have simplified the memberTotalCount to take directly the memberResponse.pagination.totalCount. For this example, I don't think I'll refactor the "totalCount" logic because it's used only once and is relatively simple to read. Thanks for the links to the resources!

Reply:
No worries Great work!